pubsub/gcloud/aio/pubsub/subscriber_client.py

```diff
@@ -1,191 +1,162 @@
+import json
+import os
 from typing import Any
-from typing import Callable
+from typing import AnyStr
 from typing import Dict
+from typing import IO
 from typing import List
 from typing import Optional
-from typing import Tuple
 from typing import Union
 
+from gcloud.aio.auth import AioSession
 from gcloud.aio.auth import BUILD_GCLOUD_REST  # pylint: disable=no-name-in-module
-from google.api_core import exceptions
-from google.cloud.pubsub_v1.subscriber.futures import StreamingPullFuture
-from google.cloud.pubsub_v1.subscriber.message import Message
-from google.cloud.pubsub_v1.types import FlowControl as _FlowControl
+from gcloud.aio.auth import Token
 
 from .subscriber_message import SubscriberMessage
 
-
-class FlowControl:
-    def __init__(self, *args: List[Any], **kwargs: Dict[str, Any]) -> None:
+if BUILD_GCLOUD_REST:
+    from requests import Session
+else:
+    from aiohttp import ClientSession as Session  # type: ignore[no-redef]
+
+API_ROOT = 'https://pubsub.googleapis.com'
+VERIFY_SSL = True
+
+SCOPES = [
+    'https://www.googleapis.com/auth/pubsub'
+]
+
+PUBSUB_EMULATOR_HOST = os.environ.get('PUBSUB_EMULATOR_HOST')
+if PUBSUB_EMULATOR_HOST:
+    API_ROOT = f'http://{PUBSUB_EMULATOR_HOST}'
+    VERIFY_SSL = False
+
+
+class SubscriberClient:
+    def __init__(self, *,
+                 service_file: Optional[Union[str, IO[AnyStr]]] = None,
+                 token: Optional[Token] = None,
+                 session: Optional[Session] = None) -> None:
+        self.session = AioSession(session, verify_ssl=VERIFY_SSL)
+        self.token = token or Token(service_file=service_file,
+                                    scopes=SCOPES,
+                                    session=self.session.session)
+
+    async def _headers(self) -> Dict[str, str]:
+        default_headers = {
+            'Content-Type': 'application/json'
+        }
+        if PUBSUB_EMULATOR_HOST:
+            return default_headers
+
+        token = await self.token.get()
+        return {
+            **default_headers,
+            'Authorization': f'Bearer {token}'
+        }
+
+    # https://cloud.google.com/pubsub/docs/reference/rest/v1/projects.subscriptions/create
+    async def create_subscription(self,
+                                  subscription: str,
+                                  topic: str,
+                                  *,
+                                  session: Optional[Session] = None,
+                                  **kwargs: Dict[str, Any]) -> Dict[str, Any]:
```

Comment:
Rather than catch-all passing the splatted `kwargs` through to the request body, it'd probably be better to be explicit on this argument; perhaps something like `body: Dict[str, Any]` or `config`, etc? Otherwise, folks will need to manually serialize/deserialize their request between the dictionary form mentioned in the API docs and Python argument form, which in many cases might involve some ugly parsing. Additionally, this may swallow more arguments than we'd like in the future, if we augment this method signature -- it'd be difficult to add, say, a `timeout` argument to this method without running into version conflicts where old versions try to use the `timeout` value as part of the actual request body!

Eg. by writing `**kwargs` here, you're only accepting arguments of the form `.create_subscription(..., foo='foo', bar='bar'), rather than the more JSON-API form of `{'foo': 'foo', 'bar': 'bar'}`.